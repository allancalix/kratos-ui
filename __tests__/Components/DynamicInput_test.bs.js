// Generated by ReScript, PLEASE EDIT WITH CARE
'use strict';

var Jest = require("@glennsl/bs-jest/src/jest.bs.js");
var Curry = require("rescript/lib/js/curry.js");
var React = require("react");
var JestDom = require("bs-jest-dom/src/JestDom.bs.js");
var DynamicInput = require("../../src/Components/DynamicInput.bs.js");
var ReactTestingLibrary = require("@rescriptbr/react-testing-library/src/ReactTestingLibrary.bs.js");

Jest.describe("DynamicInputComponent", (function (param) {
        var createComponent = function (param) {
          return React.createElement(DynamicInput.make, {
                      name: "password",
                      type: "password",
                      required: true
                    });
        };
        var createComponentWithLabel = function (param) {
          return React.createElement(DynamicInput.make, {
                      name: "password",
                      type: "password",
                      label: {
                        context: undefined,
                        id: 10,
                        text: "Cool Label",
                        type: "info"
                      },
                      required: true
                    });
        };
        Jest.test("when label is provided then matches expected", (function (param) {
                return Jest.Expect.toMatchSnapshot(Jest.Expect.expect(ReactTestingLibrary.render(undefined, undefined, undefined, undefined, undefined, createComponentWithLabel(undefined)).container));
              }));
        Jest.test("when no label is provided then matches snapshot", (function (param) {
                return Jest.Expect.toMatchSnapshot(Jest.Expect.expect(ReactTestingLibrary.render(undefined, undefined, undefined, undefined, undefined, createComponent(undefined)).container));
              }));
        return Jest.test("when label is provided then has visible label", (function (param) {
                      var partial_arg = {
                        NAME: "Str",
                        VAL: "Cool Label"
                      };
                      var arg = function (param) {
                        return JestDom.toHaveTextContent(partial_arg, param);
                      };
                      var partial_arg$1 = {
                        NAME: "Str",
                        VAL: "label"
                      };
                      var arg$1 = function (param, param$1) {
                        return ReactTestingLibrary.getByTestId(partial_arg$1, param, param$1);
                      };
                      return Curry._2(arg, undefined, expect(Curry._2(arg$1, undefined, ReactTestingLibrary.render(undefined, undefined, undefined, undefined, undefined, createComponentWithLabel(undefined)))));
                    }));
      }));

/*  Not a pure module */
