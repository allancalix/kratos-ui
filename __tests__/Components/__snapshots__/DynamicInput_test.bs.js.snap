// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicInputComponent when type is hidden then matches snapshot 1`] = `
<div>
  <input
    class="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
    name="password"
    placeholder="password"
    required=""
    type="hidden"
    value=""
  />
</div>
`;

exports[`DynamicInputComponent when type is not hidden type then matches expected 1`] = `
<div>
  <label
    class="sr-only"
    data-testid="label"
  >
    password
  </label>
  <input
    class="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
    name="password"
    placeholder="password"
    required=""
    type="password"
    value=""
  />
</div>
`;
